src/Plutus/Certification/Web3StorageClient.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module Plutus.Certification.Web3StorageClient where
+
+import Data.Aeson
+import Network.HTTP.Client      hiding (Proxy)
+import Network.HTTP.Client.TLS
+import Control.Monad.IO.Class 
+import Network.HTTP.Types.Header
+import Data.ByteString.Char8 as BS hiding (hPutStrLn)
+import Servant.Client.Core hiding (responseBody,Response,DecodeFailure)
+import Data.Text as Text
+import Servant.API hiding (addHeader)
+import Network.HTTP.Types.Status
+import Network.HTTP.Client.MultipartFormData
+import System.IO.Temp
+
+import qualified Data.ByteString.Lazy.Char8 as BSL 
+
+import GHC.IO.Handle
+type instance AuthClientData (AuthProtect "api-key") = ByteString
+
+addAuth :: ByteString -> AuthenticatedRequest (AuthProtect "api-key")
+addAuth = flip mkAuthenticatedRequest (\v -> addHeader hAuthorization ("Bearer " <> BS.unpack v))
+
+data UploadResponse = UploadResponse
+  { uploadCid :: Text
+  , uploadCarCid :: Text
+  } deriving Show
+
+instance FromJSON UploadResponse where
+  parseJSON = withObject "UploadResponse" $ \o -> UploadResponse
+    <$> o .: "cid"
+    <*> o .: "carCid"
+
+--TODO: this should be replaced with a configured key.
+apiKey :: ByteString
+apiKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJkaWQ6ZXRocjoweDE5ZWJFN0VFNUM3MUE0NUEyOEYzYjlCMjNFNzIwM0Y4MThmNGUyNmUiLCJpc3MiOiJ3ZWIzLXN0b3JhZ2UiLCJpYXQiOjE2NzA0MzU3MTg4MzAsIm5hbWUiOiJJT0cifQ.J3xzOYClx7qcOrTZErsfuMs9U9RchINAUJNzyEInsGQ"
```

Comment:
Is this safe to have in git?

Reply:
That specific key is not sensitive in itself since it is a free account I created for myself and only for development purposes.
Though I totally agree with you about the method of embedding it into the repo, and for sure it's not what I have in mind for the future. But for the moment we don't have a way to manage configurations other than the CLI arguments. 

I don't have strong opinions about how to proceed in the short term, other than the Front End it's waiting for this feature.
I'm totally open to any suggestions, some of these might be:
- we use just the CLI arguments for this key as well (the issue with this is that for the development it might be hard to always pass this long key into the command line, whenever you want to run the service )
- should we add some extra configuration method (e.g YAML file or ENV_VARS)?
- keep it like this and decide on a future task?
- other approach

Reply:
I tend toward a config file moving forward. Fine for now since the key is already published.